test/test_core.py

```diff
@@ -1972,6 +1972,24 @@ def test_main_thread_em_asm(self, args):
   def test_main_thread_async_em_asm(self, args, force_c=False):
     self.do_core_test('test_main_thread_async_em_asm.cpp', emcc_args=args, force_c=force_c)
 
+  @parameterized({
+    '': (['-sASSERTIONS']),
+    'pthreads': (['-pthread', '-sPROXY_TO_PTHREAD', '-sEXIT_RUNTIME']),
+  })
+  def test_main_thread_async_em_asm_await(self, args):
+    if '-sPROXY_TO_PTHREAD' not in args:
+      # expect runtime to error
+      output = self.do_runf('core/test_main_thread_async_em_asm_await.cpp', assert_returncode=NON_ZERO, emcc_args=args)
+      self.assertContained('emscripten_asm_const_int_await_on_main_thread is not available on the main thread', output)
+    else:
+      self.do_core_test('test_main_thread_async_em_asm_await.cpp', emcc_args=args, force_c=False)
+
+  @parameterized({
+    'pthreads': (['-pthread', '-sPROXY_TO_PTHREAD', '-sEXIT_RUNTIME', '-sASSERTIONS'], False),
+  })
```

Comment:
I think you can just remove the `@parameterized` block since there is only one variant of this test.